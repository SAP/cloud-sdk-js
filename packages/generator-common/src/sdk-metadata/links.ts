import type { Links } from './sdk-metadata-types';

export function getLinks(
  apiHubTutorialUrl: string,
  generationManualUrl: string
): Links {
  return {
    sdkDocumentation: {
      url: 'https://sap.github.io/cloud-sdk/',
      name: 'SDK Documentation'
    },
    featureDocumentation: {
      name: 'Feature Documentation',
      url:
        'https://sap.github.io/cloud-sdk/docs/js/features/odata/use-odata-v2-type-safe-client-for-javascript-typescript'
    },
    support: {
      name: 'Support',
      url: 'https://github.com/SAP/cloud-sdk-js/issues/new/choose'
    },
    apiHubTutorial: {
      name: 'API Hub Tutorial',
      url: apiHubTutorialUrl
    },
    generationManual: {
      name: 'Generation Manual',
      url: generationManualUrl
    }
  };
}
