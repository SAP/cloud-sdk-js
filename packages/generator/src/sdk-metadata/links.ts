import { Links } from './sdk-metadata-types';

export function getLinks(): Links {
  return {
    sdkDocumentation: 'https://sap.github.io/cloud-sdk/',
    featureDocumentation:
      'https://sap.github.io/cloud-sdk/docs/js/features/odata/use-odata-v2-type-safe-client-for-javascript-typescript',
    support: 'https://github.com/SAP/cloud-sdk-js/issues/new/choose',
    apiHubTutorial:
      'https://sap.github.io/cloud-sdk/docs/js/features/odata/execute-odata-request',
    generationManual:
      'https://sap.github.io/cloud-sdk/docs/js/features/odata/generate-odata-client'
  };
}
