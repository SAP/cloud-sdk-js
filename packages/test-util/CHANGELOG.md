# @sap-cloud-sdk/test-util

## 2.5.0

### Patch Changes

- Updated dependencies [89f1c423]
  - @sap-cloud-sdk/util@2.5.0
  - @sap-cloud-sdk/connectivity@2.5.0

## 2.4.0

### Patch Changes

- Updated dependencies [04726a35]
- Updated dependencies [0a008674]
- Updated dependencies [0a008674]
- Updated dependencies [0a008674]
  - @sap-cloud-sdk/connectivity@2.4.0
  - @sap-cloud-sdk/util@2.4.0
