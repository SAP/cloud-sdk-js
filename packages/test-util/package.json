{
  "name": "@sap-cloud-sdk/test-util",
  "version": "1.32.1",
  "description": "SAP Cloud SDK for JavaScript test utilities",
  "homepage": "https://community.sap.com/topics/cloud-sdk",
  "license": "Apache-2.0",
  "keywords": [
    "sap-cloud-sdk",
    "cloud-sdk",
    "sap-cloud-platform",
    "test-util"
  ],
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "dist/**/*.js",
    "dist/**/*.js.map",
    "dist/**/*.d.ts",
    "dist/**/*.d.ts.map"
  ],
  "repository": "github:SAP/cloud-sdk",
  "scripts": {
    "compile": "yarn tsc -b",
    "prepare": "yarn compile",
    "test": "yarn jest --coverage --runInBand",
    "test:local": "yarn jest",
    "test:debug": "node --inspect-brk ../../node_modules/.bin/jest --runInBand",
    "check:dependencies": "depcheck ."
  },
  "dependencies": {
    "@sap-cloud-sdk/core": "^1.32.1",
    "@sap-cloud-sdk/util": "^1.32.1"
  }
}
