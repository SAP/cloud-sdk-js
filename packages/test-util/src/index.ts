/**
 * [[include:test-util/README.md]]
 * @packageDocumentation
 * @module @sap-cloud-sdk/test-util
 */

export * from './test-destination-provider';
export * from './test-destination-mocker';
