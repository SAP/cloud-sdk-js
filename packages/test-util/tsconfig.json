{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./dist",
    "tsBuildInfoFile": "./dist/.tsbuildinfo",
    "composite": true
  },
  "include": ["src/**/*.ts"],
  "exclude": [
    "dist/**/*",
    "./test/test/**/*",
    "**/*.spec.ts",
    "node_modules/**/*"
  ],
  "references": [{ "path": "../connectivity" }]
}
