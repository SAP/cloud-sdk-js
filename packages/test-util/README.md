<!-- sap-cloud-sdk-logo -->
<!-- This block is inserted by scripts/replace-common-readme.ts. Do not adjust it manually. -->
<a href="https://sap.github.io/cloud-sdk/docs/js/overview"><img src="https://help.sap.com/doc/2324e9c3b28748a4ae2ad08166d77675/1.0/en-US/logo-with-js.svg" alt="SAP Cloud SDK for JavaScript Logo" height="122.92" width="226.773"/></a>
<!-- sap-cloud-sdk-logo-stop -->

# @sap-cloud-sdk/test-util

This package contains utility functions for testing, like loading credentials or creating test destinations.

## Installation

```
$ npm install @sap-cloud-sdk/test-util
```

## Usage

The test-util package makes writing tests for your SAP Business Technology Platform application more convenient.

For example, you can create a mock destination for your tests by using the function `mockTestDestination`.

```
import { mockTestDestination } from '@sap-cloud-sdk/test-util';
mockTestDestination('mockDestinationName');
```

You need to save your system information in local file `systems.json`:

```json
{
  "systems": [
    {
      "alias": "mockDestinationName",
      "uri": "https://www.example.com"
    }
  ]
}
```

and `credentials.json`:

```json
{
  "credentials": [
    {
      "alias": "SYS_001",
      "username": "username",
      "password": "password"
    }
  ]
}
```

<!-- sap-cloud-sdk-common-readme -->
<!-- This block is inserted by scripts/replace-common-readme.ts. Do not adjust it manually. -->
## Support

The recommended way to get in touch with us is to create an issue on [GitHub](https://github.com/SAP/cloud-sdk-js/issues).
Select the issue category **Bug**, **Feature**, or **Question** depending on the nature of your request.
We try to provide fixes, features and answers as soon as possible.

## Contribute

If you would like to contribute to the SAP Cloud SDK, please make yourself familiar with our [contributing guidelines](https://github.com/SAP/cloud-sdk-js/blob/main/CONTRIBUTING.md) and follow the given instructions.

## Links

- [Official support channel](https://github.com/SAP/cloud-sdk-js/issues/new/choose)
- [Github](https://github.com/SAP/cloud-sdk-js)
- [SAP Cloud SDK Documentation portal](https://sap.github.io/cloud-sdk)
  - [SAP Cloud SDK Documentation portal - Overview](https://sap.github.io/cloud-sdk/docs/js/overview)
  - [SAP Cloud SDK Documentation portal - Getting started guide](https://sap.github.io/cloud-sdk/docs/js/getting-started)
  - [SAP Cloud SDK Documentation portal - API documentation](https://sap.github.io/cloud-sdk/api/latest)
  - [SAP Cloud SDK Documentation portal - Release notes](https://sap.github.io/cloud-sdk/docs/js/release-notes)
- [Sample repository](https://github.com/SAP-samples/cloud-sdk-js)

## License

The SAP Cloud SDK is released under the [Apache License Version 2.0.](http://www.apache.org/licenses/)
<!-- sap-cloud-sdk-common-readme-stop -->
