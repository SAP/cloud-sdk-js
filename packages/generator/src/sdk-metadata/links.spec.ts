import { checkUrlExists } from '@sap-cloud-sdk/test-util';
import { getLinks } from './links';

describe('links',()=>{
  it('[E2E] contains only existing links',async ()=>{
    const links = getLinks();
    for(const link of Object.values(links)){
      await checkUrlExists(link);
  }
},10000);
});

