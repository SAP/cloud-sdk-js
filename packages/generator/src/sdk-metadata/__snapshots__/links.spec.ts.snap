// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sdk metadata links fails for non existing link 1`] = `"Request to https://sap.github.io/cloud-sdk/docs/js/features/non-existing-features failed."`;
